source/developers/development_validation.markdown

```diff
@@ -28,6 +28,15 @@ Some things to keep in mind:
 - Use the constants defined in `const.py`
 - Import `PLATFORM_SCHEMA` from the parent component and extend it
 - Preferred order is `required` first and `optional` second
+- Starting with Home Assistant version 0.64.0 voluptuous now requires default values for config keys to be valid values. For instance:
+```
+vol.Required(CONF_SWITCH, default=None): cv.string,
```

Comment:
If a value is required then there shouldn't be a default. We want to fail if the value is missing.

Reply:
Good point, but it was allowed with the previous version of Voluptuous. So the comment should be to remove the default parameter for vol.required but also insure that vol.optional has valid values.